components/server/src/iam/iam-session-app.spec.ts

```diff
@@ -231,6 +237,58 @@ class TestIamSessionApp {
         expect(this.teamDbMock.memberships.has(BUILTIN_INSTLLATION_ADMIN_USER_ID)).to.be.false;
         expect(this.teamDbMock.memberships.has("id-new-user")).to.be.true;
     }
+
+    @test public async testSessionRequest_updates_existing_user() {
+        const payload: OIDCCreateSessionPayload = { ...this.payload };
+        payload.claims.sub = this.knownSubjectID; // `userServiceMock.findUserForLogin` will match this value
+
+        this.knownUser.identities = [
+            {
+                authId: payload.claims.sub,
+                authProviderId: payload.claims.aud,
+                authName: "Test User",
+                primaryEmail: "random-user@any.org",
+            },
+        ];
+
+        let newEmail: string | undefined;
+        this.userServiceMock.updateUserIdentity = async (user, updatedIdentity) => {
+            newEmail = updatedIdentity.primaryEmail;
+        };
+
+        const result = await request(this.app.create())
+            .post("/session")
+            .set("Content-Type", "application/json")
+            .send(JSON.stringify(payload));
+
+        expect(result.statusCode, JSON.stringify(result.body)).to.equal(200);
+        expect(newEmail, "update was not called").not.to.be.undefined;
+        expect(newEmail).to.equal(payload.claims.email);
```

Comment:
The first check seems redundant, given we also do the second comparison, no?

Reply:
First is checking if the update is actually called, second is checking if the new value is as expected.